### PR TITLE
Fix `sticky-discussion-sidebar`

### LINKED ISSUE #2313: `sticky-discussion-sidebar` feature broken
The `.discussion-sidebar` class doesn't exist anymore, will need to use `#partial-discussion-sidebar` to target the sidebar element instead.

image

### PR DESCRIPTION
**Pull request by sharkykh**
Fixes #2313 

> Actually, it looks like they made our job easier with this clean-up.

Turns out that not really, but it's still doable.

### Test
Any discussion.

### CODE REVIEW

File: source/features/sticky-discussion-sidebar.css [hunk 1]
```diff
@@ -2,7 +2,7 @@
 Set the sidebar container's height to make it fit its contents
 (instead of being the full height of the discussion)
 */
-.discussion-sidebar {
+#discussion_bucket > :last-child {
```
Comment (sharkykh) [inline source/features/sticky-discussion-sidebar.css]: This points at the same element as the JavaScript file does. Since there is no way to get the parent element in CSS, I used this.

Comment (notlmn) [reply source/features/sticky-discussion-sidebar.css]: I think this was the layout change that made them remove the class that existed before.

This works for now, but we need to mention what the selector targets, preferably in the comments above the selector.

Comment (sharkykh) [reply source/features/sticky-discussion-sidebar.css]: It does say it targets the "sidebar container" above it, if it's not good enough I'll add a sentence similar to my comment above.

Comment (notlmn) [reply source/features/sticky-discussion-sidebar.css]: That comment does make sense, but I was referring to maybe adding a comment on why that specific CSS selector was used. Doesn't make much of a difference though, but I thought it would a little bit more clear.

Comment (sharkykh) [reply source/features/sticky-discussion-sidebar.css]: Eh, it doesn't hurt to explain a little more. Thanks you :)

File: source/features/sticky-discussion-sidebar.css [hunk 2]
```diff
@@ -1,8 +1,9 @@
 /*
 Set the sidebar container's height to make it fit its contents
 (instead of being the full height of the discussion)
+`#discussion_bucket` holds two columns: the conversation container (1st) and the sidebar container (2nd)
 */
-.discussion-sidebar {
+#discussion_bucket > :last-child {
```
Comment (fregante) [inline source/features/sticky-discussion-sidebar.css]: Why this change? `#partial-discussion-sidebar` should be enough

File: source/features/sticky-discussion-sidebar.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import debounce from 'debounce-fn';
 import features from '../libs/features';
 
 function updateStickiness(): void {
-	const sidebar = select<HTMLElement>('.discussion-sidebar')!;
+	const sidebar = select<HTMLElement>('#partial-discussion-sidebar')!.parentElement!;
```
Comment (fregante) [inline source/features/sticky-discussion-sidebar.tsx]: Same here

Comment (sharkykh) [reply source/features/sticky-discussion-sidebar.tsx]: Again, that's what I thought at first too, unfortunately no,
as it leads to this issue again:
https://github.com/sindresorhus/refined-github/pull/2294#issuecomment-517205362

nS24XMIXGj

File: source/features/sticky-discussion-sidebar.css (hunk 2 above)
Comment (sharkykh) [reply source/features/sticky-discussion-sidebar.css]: That's what I thought at first, but it seems GitHub didn't really change anything other than removing the `.discussion-sidebar` element that last week used to contain the `#partial-discussion-sidebar`.

`#partial-discussion-siderbar`` is at the correct height,
it's the container element (the column element) that needs its height adjusted.

Comment (sharkykh) [reply source/features/sticky-discussion-sidebar.css]: image

Comment (fregante) [reply source/features/sticky-discussion-sidebar.css]: Right, I see the issue. I think we should just drop it, it doesn't improve the design much and it looks like an afterthought for them too, since it's out of place.

The whole CSS would be:

```css
.rgh-sticky-sidebar {
	position: sticky;
	top: 60px !important; /* Matches sticky header's height */
	z-index: 26 !important;
}

/* Hides the last divider (on pull requests) */
.rgh-sticky-sidebar + .border-top:empty {
	display: none;
}
```

And the code below would be
```js
	const sidebar = select<HTMLElement>('#partial-discussion-sidebar')!;
```
